wire-runtime-swift/src/main/swift/ProtoCodable/WriteBuffer.swift

```diff
@@ -36,7 +36,7 @@ final class WriteBuffer {
     init(capacity: Int = 0) {
         self.capacity = 0
 
-        if capacity > 0 {
+        if capacity >= 0 {
```

Comment:
This was required to ensure `storage` was allocated in all cases (see `testAppendEmptyFirst`)

Reply:
I'm a little wary of calling `malloc(0)` in `expand()`. In C at least, the behavior for this [can vary](https://stackoverflow.com/questions/2132273/what-does-malloc0-return) from runtime to runtime.

Can we achieve the same thing by adding something like this into `append`:
```
guard !data.isEmpty else { return }
```

Reply:
Added `guard`s where appropriate